staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -122,6 +122,26 @@ type JSONSchemaProps struct {
 	//      Atomic maps will be entirely replaced when updated.
 	// +optional
 	XMapType *string
+
+	// x-kubernetes-validations describes a list of validation rules for expression validation.
+	// This field is an alpha-level. Using this field requires the feature gate `CustomResourceValidationExpressions` to be enabled.
+	XValidations ValidationRules
+}
+
+// ValidationRules describe a list of validation rules for expression validation
+//  +listType=atomic
+type ValidationRules []ValidationRule
+
+// ValidationRule is used for expression validation.
+type ValidationRule struct {
+	// Rule represents the validation rule which will be evaluated by CEL.
+	// ref: https://github.com/google/cel-spec
+	// The validator will be scoped to the location of the x-kubernetes-validations extension in the schema
+	// and will use `self` to represent the scoped field name.
+	// If an object property collides with a RESERVED keyword, it will be escaped by prepending a _ prefix.
```

Comment:
we have to list all reserved keywords and this list must never change in the future.

Reply:
Can't it be a link to https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax?

Reply:
Added the link. Not sure if "this list must never change in the future". Could we ask user refer to the doc?

Reply:
It would be very hard to change the list in the future. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#backward-compatibility-gotchas

Are all the restricted words part of the CEL documentation or do we have extra words?